packages/engine/lib/memory/src/arrow/flush/mod.rs

```diff
@@ -1,38 +1,48 @@
-use arrow::{array::ArrayData, buffer::Buffer, util::bit_util};
-
 use crate::{
-    arrow::meta,
+    arrow::{meta, util::bit_util},
     error::Result,
     shared_memory::{padding, BufferChange, Segment},
 };
 
-/// The info required about Arrow array data in order to grow it
+mod arrow2_growable_array_data;
+
+/// This is an interface which provides a way to extract the data we need about an Arrow array
+/// in order to be able to grow it.
 ///
 /// Can be implemented by ArrayData, ArrayDataRef, Python FFI array data.
 pub trait GrowableArrayData: Sized + std::fmt::Debug {
     fn len(&self) -> usize;
     fn is_empty(&self) -> bool {
         self.len() == 0
     }
+    /// Returns the number of items which are null in the given array.
     fn null_count(&self) -> usize;
+    /// This returns a bitmap, where the nth bit of the returned data specifies whether or not the
+    /// nth item in the array is null or not.
     fn null_buffer(&self) -> Option<&[u8]>;
+    /// Returns the nth buffer of this array. We follow the
+    /// [specification](https://arrow.apache.org/docs/format/Columnar.html#buffer-listing-for-each-layout)
+    /// _except_ that we don't return any validity bitmaps/null buffers here (these can be obtained
+    /// instead by calling [`GrowableArrayData::null_buffer`]).
+    #[allow(clippy::redundant_allocation)]
     fn buffer(&self, index: usize) -> &[u8];
     /// Arrow stores the null buffer separately from other buffers.
     fn non_null_buffer_count(&self) -> usize;
+    /// This returns the data of the child arrays.
     fn child_data(&self) -> &[Self];
 }
 
-impl GrowableArrayData for ArrayData {
+impl GrowableArrayData for arrow::array::ArrayData {
```

Comment:
Do we expect to remove this? I think this was very helpful for testing the new implementation, but is there a benefit in keeping it?

Reply:
Awesome, thanks!